src/engine/MaxConstraint.cpp

```diff
@@ -492,12 +512,32 @@ bool MaxConstraint::constraintObsolete() const
     return _obsolete;
 }
 
-void MaxConstraint::eliminateVariable( unsigned var, double /*value*/ )
-{
+void MaxConstraint::eliminateVariable( unsigned var, double value )
+{ // todo updated - Guy A
+    if ( !_eliminatedSomeVariables )
```

Comment:
clearner: set eliminated flag to true. max is the max of new value and old (initialize to -infinity).